gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -159,6 +159,114 @@ def import_spliceai_vcf(path: str) -> hl.Table:
     return ht
 
 
+def create_pangolin_grch38_ht() -> hl.Table:
+    """
+    Create a Hail Table with Pangolin score for splicing for GRCh38.
+
+    .. note::
+    The score was based on the splicing prediction tool Pangolin:
+    Zeng, T., Li, Y.I. Predicting RNA splicing from DNA sequence using Pangolin.
+     Genome Biol 23, 103 (2022). https://doi.org/10.1186/s13059-022-02664-4
+
+    There's no precomputed score for all possible variants, the scores were
+    generated for gnomAD v4 genomes (=v3 genomes) and v4 exomes variants in
+    gene body only.
+
+    :return: Hail Table with Pangolin score for splicing for GRCh38.
+    """
+    v4_genomes = (
+        "gs://gnomad-insilico/pangolin/gnomad.v4.0.genomes.pangolin.vcf.bgz/*.bgz"
+    )
+    v4_exomes = "gs://gnomad-insilico/pangolin/gnomad.v4.0.exomes.pangolin.vcf.bgz/*.gz"
+
+    def import_pangolin_vcf(vcf_path: str) -> hl.Table:
+        """
+        Import Pangolin VCF to Hail Table.
+
+        :param vcf_path: Path to Pangolin VCF.
+        :return: Hail Table with Pangolin scores.
+        """
+        ht = hl.import_vcf(
+            vcf_path,
+            min_partitions=1000,
+            force_bgz=True,
+            reference_genome="GRCh38",
+            skip_invalid_loci=True,
+        ).rows()
+        return ht
+
+    ht_g = import_pangolin_vcf(v4_genomes)
+    ht_e = import_pangolin_vcf(v4_exomes)
+
+    logger.info(
+        "Number of rows in genomes Pangolin HT: %s; "
+        "Number of rows in raw exomes Pangolin HT: %s. ",
+        ht_g.count(),
+        ht_e.count(),
+    )
+
+    ht = ht_g.union(ht_e)
+
+    logger.info("Exploding Pangolin scores...")
+    # `explode` will eliminate rows with empty array
+    # The VCF INFO of Pangolin is a string with the following format:
+    # gene1|pos_splice_gain:largest_increase|pos_splice_loss:largest_decrease|gene2...
+    # for example:
+    # Pangolin=ENSG00000121005.9|-86:0.25|38:-0.49|Warnings:||ENSG00000254238.1|-40:0.01|30:-0.17|Warnings:
+
+    ht = ht.annotate(pango=ht.info.Pangolin[0].split(delim="\|\|"))
```

Comment:
Because of error: 'ArrayExpression' object has no attribute 'split', if removing [0]. It's an array of strings, and there's only one string element before splitting. I will write a note.